test/http-client.spec.js

```diff
@@ -787,20 +784,49 @@ describe('http client', () => {
         });
     });
 
-    it('should be applied after all transformers', (done) => {
-      class RequestInterceptor {
+    it('should be possible to modify a header in a request interceptor', (done) => {
+      class Interceptor {
         request(message) {
+          message.headers.add('aheader', 'anothervalue');
+          return message;
+        }
+
+        response(message) {
+          var request = jasmine.Ajax.requests.mostRecent();
+          expect(request.requestHeaders['aheader']).toBe('anothervalue')
+          return message;
+        }
+      }
+
+      var interceptor = new Interceptor();
+      var client = new HttpClient()
+        .configure(x => {
+          x.withInterceptor(interceptor);
+          x.withBaseUrl(baseUrl);
+        });
+
+      client.createRequest('some/cool/path')
+        .asGet()
+        .send()
+        .then((response) => {
+          done();
+        });
+    });
+
+    it('should apply xhr transformers before response interceptors are called', (done) => {
```

Comment:
No, I just mean a test that explicitly verifies the change you made to run request interceptors before the xhrTransformers. Something that would break if that ordering were reversed. I suppose the headers test above would break, though, so maybe it's unnecessary.